File: poo/primero/libreriaJava/1/employee.java
public class employee {
    public static void main(String[] args) {
        EmployeeSalary employeeSalary = new EmployeeSalary("Laura Vanessa", 2000000, 7, 4, 2);
        Subsidy subsidy = new Subsidy(1300000);
        OvertimePay overtimePay = new OvertimePay();
        Deductions deductions = new Deductions("medium");
        Salary salary = new Salary(employeeSalary, subsidy, overtimePay, deductions);

        System.out.println("Nombre: " + employeeSalary.getEmployeeName());
        System.out.println("Salario base: " + String.format("%,d", (int) employeeSalary.getBaseSalary()));
        System.out.println("Horas extras diurnas: " + employeeSalary.getDaytimeOvertimeHours());
        System.out.println("Horas extras nocturnas: " + employeeSalary.getNighttimeOvertimeHours());
        System.out.println("Horas extras festivas: " + employeeSalary.getHolidayOvertimeHours());
        System.out.println("Subsidio: " + String.format("%,d", (int) subsidy.calculateSubsidy(employeeSalary)));
        System.out.println("Pago horas extra: " + String.format("%,d", (int) overtimePay.calculateOvertimePay(employeeSalary)));
        System.out.println("Deducciones: " + String.format("%,d", (int) deductions.calculateDeductions(employeeSalary)));
        System.out.println("Salario total neto: " + String.format("%,d", (int) salary.calculateSalary()));
    }
}

class EmployeeSalary {
    private String employeeName;
    private double baseSalary;
    private int daytimeOvertimeHours;
    private int nighttimeOvertimeHours;
    private int holidayOvertimeHours;

    public EmployeeSalary(String employeeName, double baseSalary, int daytimeOvertimeHours, int nighttimeOvertimeHours, int holidayOvertimeHours) {
        this.employeeName = employeeName;
        this.baseSalary = baseSalary;
        this.daytimeOvertimeHours = daytimeOvertimeHours;
        this.nighttimeOvertimeHours = nighttimeOvertimeHours;
        this.holidayOvertimeHours = holidayOvertimeHours;
    }

    public String getEmployeeName() {
        return employeeName;
    }

    public double getBaseSalary() {
        return baseSalary;
    }

    public int getDaytimeOvertimeHours() {
        return daytimeOvertimeHours;
    }

    public int getNighttimeOvertimeHours() {
        return nighttimeOvertimeHours;
    }

    public int getHolidayOvertimeHours() {
        return holidayOvertimeHours;
    }
}

class Subsidy {
    private double minimumWage;

    public Subsidy(double minimumWage) {
        this.minimumWage = minimumWage;
    }

    public double calculateSubsidy(EmployeeSalary employeeSalary) {
        if (employeeSalary.getBaseSalary() < 2 * minimumWage) {
            return minimumWage;
        } else {
            return 0;
        }
    }
}

class OvertimePay {
    public double calculateOvertimePay(EmployeeSalary employeeSalary) {
        double base = employeeSalary.getBaseSalary() / 240;
        double daytimeRate = base * 1.25;
        double nighttimeRate = base * 1.75;
        double holidayRate = base * 2.0;

        double daytimePay = employeeSalary.getDaytimeOvertimeHours() * daytimeRate;
        double nighttimePay = employeeSalary.getNighttimeOvertimeHours() * nighttimeRate;
        double holidayPay = employeeSalary.getHolidayOvertimeHours() * holidayRate;

        return daytimePay + nighttimePay + holidayPay;
    }
}

class Deductions {
    private String arlRiskLevel;

    public Deductions(String arlRiskLevel) {
        this.arlRiskLevel = arlRiskLevel;
    }

    public double calculateDeductions(EmployeeSalary employeeSalary) {
        double salary = employeeSalary.getBaseSalary();
        double health = salary * 0.04;
        double pension = salary * 0.04;
        double arl = salary * getArlPercentage();

        return health + pension + arl;
    }

    private double getArlPercentage() {
        return switch (arlRiskLevel.toLowerCase()) {
            case "low" -> 0.02;
            case "medium" -> 0.03;
            case "high" -> 0.04;
            default -> 0.0;
        };
    }
}

class Salary {
    private EmployeeSalary employeeSalary;
    private Subsidy subsidy;
    private OvertimePay overtimePay;
    private Deductions deductions;

    public Salary(EmployeeSalary employeeSalary, Subsidy subsidy, OvertimePay overtimePay, Deductions deductions) {
        this.employeeSalary = employeeSalary;
        this.subsidy = subsidy;
        this.overtimePay = overtimePay;
        this.deductions = deductions;
    }

    public double calculateSalary() {
        double totalSubsidy = subsidy.calculateSubsidy(employeeSalary);
        double totalOvertime = overtimePay.calculateOvertimePay(employeeSalary);
        double totalDeductions = deductions.calculateDeductions(employeeSalary);

        return employeeSalary.getBaseSalary() + totalSubsidy + totalOvertime - totalDeductions;
    }
}

File: poo/primero/libreriaPhp/1/employee.php

<?php
class EmployeeSalary {
    private string $employeeName;
    private float $baseSalary;
    private int $daytimeOvertimeHours;
    private int $nighttimeOvertimeHours;
    private int $holidayOvertimeHours;

    public function __construct(string $employeeName, float $baseSalary, int $daytimeOvertimeHours, int $nighttimeOvertimeHours, int $holidayOvertimeHours) {
        $this->employeeName = $employeeName;
        $this->baseSalary = $baseSalary;
        $this->daytimeOvertimeHours = $daytimeOvertimeHours;
        $this->nighttimeOvertimeHours = $nighttimeOvertimeHours;
        $this->holidayOvertimeHours = $holidayOvertimeHours;
    }

    public function getEmployeeName() {
        return $this->employeeName;
    }

    public function getBaseSalary() {
        return $this->baseSalary;
    }

    public function getDaytimeOvertimeHours() {
        return $this->daytimeOvertimeHours;
    }

    public function getNighttimeOvertimeHours() {
        return $this->nighttimeOvertimeHours;
    }

    public function getHolidayOvertimeHours() {
        return $this->holidayOvertimeHours;
    }
}

class Subsidy {
    private float $minimumWage;

    public function __construct(float $minimumWage) {
        $this->minimumWage = $minimumWage;
    }

    public function calculateSubsidy(EmployeeSalary $employeeSalary) {
        if ($employeeSalary->getBaseSalary() < 2 * $this->minimumWage) {
            return $this->minimumWage;
        } else {
            return 0;
        }
    }
}

class OvertimePay {
    public function calculateOvertimePay(EmployeeSalary $employeeSalary) {
        $daytimeOvertimeRate = $employeeSalary->getBaseSalary() / 240 * 1.25;
        $nighttimeOvertimeRate = $employeeSalary->getBaseSalary() / 240 * 1.75;
        $holidayOvertimeRate = $employeeSalary->getBaseSalary() / 240 * 2;

        $daytimeOvertimePay = $employeeSalary->getDaytimeOvertimeHours() * $daytimeOvertimeRate;
        $nighttimeOvertimePay = $employeeSalary->getNighttimeOvertimeHours() * $nighttimeOvertimeRate;
        $holidayOvertimePay = $employeeSalary->getHolidayOvertimeHours() * $holidayOvertimeRate;

        return $daytimeOvertimePay + $nighttimeOvertimePay + $holidayOvertimePay;
    }
}

class Deductions {
    private string $arlRiskLevel;

    public function __construct(string $arlRiskLevel) {
        $this->arlRiskLevel = $arlRiskLevel;
    }

    public function calculateDeductions(EmployeeSalary $employeeSalary) {
        $healthDeductions = $employeeSalary->getBaseSalary() * 0.04;
        $pensionDeductions = $employeeSalary->getBaseSalary() * 0.04;
        $arlDeductions = $employeeSalary->getBaseSalary() * $this->getArlPercentage();

        return $healthDeductions + $pensionDeductions + $arlDeductions;
    }

    private function getArlPercentage() {
        switch ($this->arlRiskLevel) {
            case 'low':
                return 0.02;
            case 'medium':
                return 0.03;
            case 'high':
                return 0.04;
            default:
                return 0;
        }
    }
}

class Salary {
    private $employeeSalary;
    private $subsidy;
    private $overtimePay;
    private $deductions;

    public function __construct(EmployeeSalary $employeeSalary, Subsidy $subsidy, OvertimePay $overtimePay, Deductions $deductions) {
        $this->employeeSalary = $employeeSalary;
        $this->subsidy = $subsidy;
        $this->overtimePay = $overtimePay;
        $this->deductions = $deductions;
    }

    public function calculateSalary() {
        $subsidy = $this->subsidy->calculateSubsidy($this->employeeSalary);
        $overtimePay = $this->overtimePay->calculateOvertimePay($this->employeeSalary);
        $deductions = $this->deductions->calculateDeductions($this->employeeSalary);

        return $this->employeeSalary->getBaseSalary() + $subsidy + $overtimePay - $deductions;
    }
}

$employeeSalary = new EmployeeSalary ('Laura vanessa', 2000000, 7, 4, 2);
$subsidy = new Subsidy (1300000);
$overtimePay = new OvertimePay ();
$deductions = new Deductions('medium');

$salary = new Salary($employeeSalary, $subsidy, $overtimePay, $deductions);


   echo "Nombre: " . $employeeSalary->getEmployeeName() . "\n";
   echo "<br/>";
   echo "Salario base: " . number_format($employeeSalary->getBaseSalary()) . "\n";
   echo "<br/>";
   echo "Horas extras diurnas: " . $employeeSalary->getDaytimeOvertimeHours() . "\n";
   echo "<br/>";
   echo "Horas extras nocturnas: " . $employeeSalary->getNighttimeOvertimeHours() . "\n";
   echo "<br/>";
   echo "Horas extras festivas: " . $employeeSalary->getHolidayOvertimeHours() . "\n";
   echo "<br/>";
   echo "Subsidio: " . number_format($subsidy->calculateSubsidy($employeeSalary)) . "\n";
   echo "<br/>";
  echo "Pago horas extra: " . number_format($overtimePay->calculateOvertimePay($employeeSalary)) . "\n";
  echo "<br/>";
  echo "Deducciones: " . number_format($deductions->calculateDeductions($employeeSalary)) . "\n";
  echo "<br/>";
  echo "Salario total neto: " . number_format($salary->calculateSalary()) . "\n";


?>